### ISSUE TITLE
Add Task case to handle request with Encodable type.

### ISSUE THREAD

**Issue by sunshinejr** [Moya/Moya#1325]
Long read: #1135. Short one: we need an option to send `Encodable` type that will be converted to data and passed to `httpBody`. It could look somehow similar to:
```swift
enum Task {
    ...
    case requestJsonEncodable(Encodable)
    ...
}
```

**Comment by afonsograca** [comment]
Hi,
I was trying to give this a go, but it seems that this requires more changes than first thought. Since `Encodable` doesn't conform to itself, when trying to encode
```swift
JSONEncoder().encode(encodable)
```
We get the following error `Cannot invoke 'encode' with an argument list of type '((Encodable))'` because it needs a concrete type to conform to the protocol.

This could be solved by defining making the `Task` enum generic, but this would involve a major breaking change, which in my opinion is not worth it. Encoding can be done prior to the request call.

**Comment by SD10** [comment]
@afonsograca Thank's for giving this a shot. There's an open PR #1326 regarding this. It will be a difficult one to work out. As you mentioned, I don't believe making the entire `Task` type generic to support a single case is the right solution.  Hopefully, I can look into this more sometime this week.

**Comment by afonsograca** [comment]
@SD10 damn, only saw your message now. I totally missed the open PR.

I ended type erasing the protocol and from the tests I've done it seems to work. I opened a PR, but given that there's already an open PR regarding this, please disregard mine. 

All in all, this was a good brain teaser :D haven't had to use type erasure in a while.
Keep up the good work!

**Comment by devxoul** [comment]
@sunshinejr Just for making it sure: did you mean `httpBody` instead of `htmlBody`?

**Comment by sunshinejr** [comment]
Right, how the hell did I think of `htmlBody` I'm not sure 😥

**Comment by sunshinejr** [comment]
Not sure why I had to close it manually as the PR had `Resolves #1325` keyword, but yeah, it is already in beta so closing this one!

**Comment by BasThomas** [comment]
GitHub's a little funny there - I know of only "fixes #whatever" that automatically closes the issue.
